src/reducers/InputOutput.js

```diff
@@ -1,9 +1,32 @@
 import NPMsAlert from 'react-s-alert';
-
+import math from 'mathjs';
 import {exportToDisk} from "../utils/downloader";
+import {Made} from "@exabyte-io/made.js";
 import {MATERIALS_EXPORT, MATERIALS_ADD, MATERIALS_REMOVE, MATERIALS_SAVE} from "../actions";
+import {Material} from "../material";
 
 export function materialsSave(state, action) {
+    const index = state.index || 0;
+    const stateItems = state.materials.slice();
+    const material = stateItems[index];
+    const newMaterial = new Material(material);
+    if (material.isNonPeriodic) {
+        const lattice = [material.Lattice.a, material.Lattice.b, material.Lattice.c];
+        const updatedBasis = new Made.Basis(material.basis);
+        const centerOfCoordiantes = updatedBasis.centerOfCoordinatesPoint;
+        const centerOfLattice = math.multiply(0.5, lattice);
+        const translationVector = math.subtract(centerOfLattice, centerOfCoordiantes);
+        updatedBasis.translateByVector(translationVector);
+        const newBasis = {
+            cordinates: updatedBasis._coordinates.array,
+            elements: updatedBasis._elements.array
+        }
+        newMaterial.isUpdated = material.isUpdated;
+        newMaterial.lattice = material.lattice;
+        newMaterial.isNonPeriodic = material.isNonPeriodic;
+        newMaterial.basis = newBasis;
+        state.materials[state.index] = newMaterial;
```

Comment:
The state is not complete. The value if `Basis` gets set to the default parameters of `[Si], [0,0,0]` instead of the values from `basis`.  Need to figure out why that is happening